scm-plugin/src/main/resources/archetype-resources/src/main/java/HelloWorldSCM.java

```diff
@@ -0,0 +1,61 @@
+package $package;
+
+import java.io.File;
+import java.io.IOException;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.scm.ChangeLogParser;
+import hudson.scm.PollingResult;
+import hudson.scm.SCM;
+import hudson.scm.SCMDescriptor;
+import hudson.scm.SCMRevisionState;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.SCMRevision;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class HelloWorldSCM extends SCM {
+    HelloWorldSCM(SCMHead head, SCMRevision revision) {
```

Comment:
Uh, no. You need a `@DataBoundConstructor`. An `SCMSource.build` will generally wish to preconfigure an `SCM` it returns, but the details will vary widely and generally you would be calling a `@DataBoundConstructor` plus perhaps some `@DataBoundSetter`s.